sig-multicluster/namespace-sameness-position-statement.md

```diff
@@ -92,10 +96,29 @@ thus namespce sameness applies), the implementation of multi-cluster services
 probably should not be on-by-default.  The details of how multi-cluster
 services will work are an area for innovation, but ideas include:
   * Opt-in: services must be "exported" to be merged across clusters
-  * Opt-out: services or namespace can be opted out of service merging
+  * Opt-out: services or namespaces can be opted out of sameness
   * Different discovery: merged services and "raw" services use different names
     or other discovery mechanisms
 
 However the implementation works, the metrics service is not automatically
 merged across clusters, though the LDAP-to-RBAC sync from example 2 can still
 apply consistent policies.
+
+### Example 4: Authority sameness policy
+
+Consider the same organization from previous examples, but with more clusters:
+A, B, C, and D.  They want to assign foo-team's namespaces to clusters A and B,
+and bar-team's namespaces to clusters C and D.  They also want to ensure that
+the opposite is never true - foo-team's namespaces cannot be used in clusters C
+or D, and bar-team's namespaces cannot be used from clusters A or B.
```

Comment:
This specific example feels like it _doesn't_ fall under the scope defined in this document.

> the set of all clusters, governed by a single authority, that are expected to work together

I'd contend that what is being described is actually two independent sets of clusters, `clusterset-foo` consisting of A and B, and `clusterset-bar` with clusters C and D, and what is needed for this scenario is not a complex "sameness" configuration, but rather **introducing a pattern for "cross-clusterset" service networking**, where sameness is _not_ assumed outside the clusterset boundary.

"Cross-clusterset" service networking would enable patterns like `foo-team` exporting an `api` service to `clusterset-bar`, where `bar-team` could route traffic from their `web` frontend to the imported `api` backend from `clusterset-foo` (a `ServiceImport` likely could not be created _automatically_ by a multicluster controller though, as it would be ambiguous in which namespace it belongs without additional configuration).

Consul service mesh currently offers similar functionality through our [cluster peering](https://developer.hashicorp.com/consul/docs/connect/cluster-peering) feature between individual clusters, and we would be interested in helping standardize a pattern for this at the clusterset abstraction layer. The desire for a cross-clusterset service networking model was also described by @srampal during a [presentation](https://docs.google.com/presentation/d/19hAg0jJO6WIyt5EyeWwpVbPNXbI9iyi7np4i9LzrNYE/edit) in recent [SIG-Multicluster](https://docs.google.com/document/d/18mk62nOXE_MCSSnb4yJD_8UadtzJrYyJxFwbrgabHe8/edit#heading=h.u9paxojwbldq) and [Gateway API GAMMA](https://docs.google.com/document/d/1s5hQU0CB9ehjFukRmRHQ41f1FA8GX5_1Rv6nHW6NWAA/edit#heading=h.x5yxcfasfg03) meetings.

Reply:
From speaking to lots of people about this, I think it's unfortunately more complicated than that.  It's not always perfectly-disjoint or perfectly-overlapping.

I'm wary of demanding perfect-sameness (I tried, believe me :) because it ends up with a large number of clustersets, each with a small number of clusters, which reduces to the same management-at-sclae problems as before.  We then have to define sets of sets, and likely sets of sets of sets.  It's UX hell all the way down.

The compromise that seems to make the most sense to people is some sort of intra-set selection - explicit or by metadata or by convention - which governs and in-group and an out-group for each question-under-consideration.  All I want to do with this PR is make sure we allow for implementations to try things without running afoul of the letter of the law.  I am NOT trying to describe an API or even a required semantic.

I do think that cross-clusterset stuff is needed, especially in networking land, but I don't think it is *this* problem.